### PR TITLE
Table Panel: Fix image disappearing when datalinks applied

### PR DESCRIPTION
**Pull request by codeincarnate**
**What is this feature?**

When using the image cell with a datalink and hovering over the image the image would disappear because no absolute width or height is applied. This makes the link functionally have zero width and height making it so it's no displayed. This PR sets the height of the link to the height of the cell (minus 2px).

With old styles container width would be sufficient for this, but with height now changing with hover for text wrapping this no longer works in the intended way.

**Why do we need this feature?**

Prevents images from disappearing on hover.

### Before: 

https://github.com/grafana/grafana/assets/199847/20312796-18f3-40a8-9fbf-96218fe32dff

### After:

https://github.com/grafana/grafana/assets/199847/7a3cde94-42b9-4b9a-8bab-233a3dd76873

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by gtk-grafana** [comment]
@codeincarnate do you have a datasource that returns image urls I can use to test?

**Review by nmarrs — COMMENTED** [review]
Tested locally and wasn't able to get a repro case working (seeing same seemingly ok behavior on both main / this branch

Is it possible for you to provide a repro dashboard? I'm not sure on the exact reproduction steps to test and verify this fix :) (i.e. does the image need to be only the size of the cell for it to disappear?)

Dashboard I tried to repro with
[Table image example-1710777486116.json](https://github.com/grafana/grafana/files/14639136/Table.image.example-1710777486116.json)


https://github.com/grafana/grafana/assets/22381771/15258597-c2a4-4a2c-ac44-9a331026fe46

**Comment by codeincarnate** [comment]
@gtk-grafana @nmarrs I'm using [this dashboard for testing](https://github.com/grafana/grafana/files/14641864/images-disappearing.json)

File: public/locales/pseudo-LOCALE/grafana.json [hunk 1]
```diff
@@ -369,7 +369,6 @@
       "title-label": "Ŧįŧľę"
     },
     "json-editor": {
-      "apply-button": "Åppľy čĥäŉģęş",
```
Comment (baldm0mma) [inline public/locales/pseudo-LOCALE/grafana.json:372]: LOL! What is this?!?!?!

Comment (codeincarnate) [reply public/locales/pseudo-LOCALE/grafana.json:372]: Translation extraction stuff, the default pseudo-locale is funky. CI was complaining about it but seems to be good now 🤔

**Review by baldm0mma — APPROVED** [review]
Like a charm!

https://github.com/grafana/grafana/assets/46619047/44d49bbf-ad84-4617-9cff-2218ff01ef85

**Review by nmarrs — APPROVED** [review]
Minor nit around cell styling on hover but otherwise confirmed fix locally :) (see blue table cell hover styling extending out of bounds with changes from this PR)

Before

https://github.com/grafana/grafana/assets/22381771/f44eacda-be1e-4883-a219-4fd98762a5ea



After


https://github.com/grafana/grafana/assets/22381771/127a87b1-1d46-4055-9748-bef92a6bbfe6

**Comment by codeincarnate** [comment]
Fixed up the issue with the height shifting. Applying an offset that's inline with the padding so nothing changes 😄

**Comment by nmarrs** [comment]
Test locally and LGTM! :shipit:

**Review by gtk-grafana — APPROVED** [review]
LGTM, working locally for me as well!

**Pull request merged**